Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -26,6 +26,35 @@ extension MongoCollection {
         return try self._client.executeOperation(operation, session: session)
     }
 
+    /**
+     * Finds a single document in this collection that matches the provided filter.
+     *
+     * - Parameters:
+     *   - filter: A `Document` that should match the query
+     *   - options: Optional `FindOneOptions` to use when executing the command
+     *   - session: Optional `ClientSession` to use when executing this command
+     *
+     * - Returns:  the resulting `Document`
+     *
+     * - Throws:
+     *   - `UserError.invalidArgumentError` if the options passed are an invalid combination.
+     *   - `UserError.logicError` if the provided session is inactive.
+     *   - `EncodingError` if an error occurs while encoding the options to BSON.
+     */
+
+    public func findOne(
+        _ filter: Document = [:],
+        options: FindOneOptions? = nil,
+        session: ClientSession? = nil
+    ) throws -> T? {
+        var findOneOptions: FindOptions?
+        if let options = options {
+            findOneOptions = FindOptions(findOneOptions: options)
+        }
+        let cursor = try self.find(_: filter, options: findOneOptions, session: session)
```

Comment:
nit: I think you can just say `filter` instead of `_: filter`. the use of that in the `find` signature is so users don't have to provide a parameter label